src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseDownloader.java

```diff
@@ -100,14 +100,14 @@ public ExerciseDownloader(UrlCommunicator urlCommunicator, TmcApi tmcApi) {
         return Optional.of(downloadedExercises);
     }
 
-    public String createCourseFolder(String path, String folderName) {
-        path = formatPath(path);
+    public Path createCourseFolder(Path path, String folderName) {
+
         if (!isNullOrEmpty(folderName)) {
-            path += folderName + File.separator;
+            path = path.resolve(folderName);
         }
-        File coursePath = new File(path);
-        if (!coursePath.exists()) {
-            coursePath.mkdirs();
+
+        if (!Files.exists(path)) {
+            path.toFile().mkdirs();
```

Comment:
Prefer [`java.nio.Files#createDirectories(Path dir, FileAttribute<?>... attrs)`](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#createDirectories(java.nio.file.Path,%20java.nio.file.attribute.FileAttribute...))  (and maybe [`java.nio.Files#notExists(Path path, LinkOption... options)`](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#notExists(java.nio.file.Path,%20java.nio.file.LinkOption...)) on the line above, although this has slightly different semantics)